service-base/src/main/java/org/eclipse/hono/service/auth/device/DeviceCredentials.java

```diff
@@ -39,4 +42,17 @@
      * @return The tenant.
      */
     String getTenantId();
+
+    /**
+     * Gets additional properties that can be used to get credentials from the Credentials API.
+     * <p>
+     * This default implementation returns an empty JSON object.
+     *
+     * @return A new JsonObject. An implementation can return a JSON object containing arbitrary properties. It is
```

Comment:
FMPOV this parameter declaration should be applicable to implementing classes as well. Thus, IMHO it should not describe the default implementation strategy but instead the abstract semantics of the object returned, e.g.
> @return The properties representing the client context.

The describing sentence *An implementation ...* should be moved to a paragraph of the method description.